src/coreclr/tools/aot/crossgen2/crossgen2.csproj

```diff
@@ -59,4 +57,19 @@
       </ItemGroup>
   </Target>
 
+  <Target Name="LocateNativeCompiler"
+          Condition="'$(_IsPublishing)' == 'true' and '$(HostOS)' != 'windows'"
```

Comment:
I think we should also check for `'$(NativeAotSupported)' == 'true'`

Reply:
Yes, thanks!
```suggestion
          Condition="'$(NativeAotSupported)' == 'true' and '$(_IsPublishing)' == 'true' and '$(HostOS)' != 'windows'"
```